src/Microsoft.ML.FastTree/Utils/ToByteArrayExtensions.cs

```diff
@@ -683,10 +689,12 @@ public static double[][] ToDoubleJaggedArray(this byte[] buffer, ref int positio
         public static long SizeInBytes(this string[] array)
         {
             long length = sizeof(int);
-            for (int i = 0; i < array.Length; ++i)
+            for (int i = 0; i < Utils.Size(array); ++i)
             {
                 length += array[i].SizeInBytes();
             }
+            if (length > 0)
```

Comment:
>if (length > 0) [](start = 12, length = 15)

Is this intentional? `long length = sizeof(int)` means that it will be initialized to 8, so it should always be positive. Was this check meant to be on `Utils.Size(array)` instead?

Reply:
Actually I'm not sure what my intention was. If the length of array is 0, then the size returned should be sizeof(int) since the length of the array is always serialized before the array itself. 

---
In reply to: [196203608](https://github.com/dotnet/machinelearning/pull/364#discussion_r196203608) [](ancestors = 196203608)